src/dash-table/derived/paginator.ts

```diff
@@ -21,15 +23,15 @@ function getBackEndPagination(
     return {
         loadNext: () => {
             pagination_settings.current_page++;
-            setProps({ pagination_settings });
+            setProps({ pagination_settings, ...clearSelection });
```

Comment:
Clearing cell selection on page changes. You can select *rows* on separate pages and have them stay selected... but that's with row-by-row checkboxes, an inherently stickier interaction that feels right to persist even when a checked box is on a hidden page.